### PR TITLE
Story Two tasks for review

### PR DESCRIPTION
**Pull request by EleanorRoberts**

### CODE REVIEW

File: addpokemon.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<?php
+require_once 'collectiondb.php';
+?>
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8" />
+    <title>Pokemon Collection</title>
+
+<!--	<link rel="stylesheet" href="normalize.css" />-->
```
Comment (Roomsycakes01) [inline addpokemon.php]: do this before the main push

File: addpokemon.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+<!--	<link rel="stylesheet" href="xxx.css" />-->
+	<meta name="viewport" content="width=device-width, initial-scale=1.0" />
+
```
Comment (Roomsycakes01) [inline addpokemon.php]: whitespace

File: addpokemon.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    <label for="defense">Defense <input type="text" name="defense" required /></label>
+    <label for="spAttack">Sp. Attack <input type="text" name="spAttack" required /></label>
+    <label for="spDefense">Sp. Defense <input type="text" name="spDefense" required/></label>
+    <label for="speed">Speed <input type="text" name="speed" required/></label>
+    <input type="submit" name="submit" />
+</form>
+</body>
+</html>
+
+<?php
```
Comment (Roomsycakes01) [inline addpokemon.php]: ?????

File: collectionFunctions.php [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<?php
+
+function addNewItem() {
+
+}
```
Comment (Roomsycakes01) [inline collectionFunctions.php]: newline

File: collectionchallenge.php [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    echo "<li>SP. Attack: {$poke['spAttack']}</li>";
+    echo "<li>SP. Defense: {$poke['spDefense']}</li>";
+    echo "<li>Speed: {$poke['speed']}</li>";
+    echo "</div>";
+}
+
+?>
+<a href="addpokemon.php">Add New Pokemon</a>
+</body>
+</html>
```
Comment (Roomsycakes01) [inline collectionchallenge.php]: newline

File: collectionchallenge.php [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+<body>
+<h1>Pokemonz</h1>
+<?php
+foreach ($pokemon as $poke) {
+    echo "<div>";
+    echo "<h2>{$poke['name']}</h2>";
+    if ($poke['type2'] === "") {
+        echo "<p>{$poke['type1']}</p>";
+    } else {
+        echo "<p>{$poke['type1']}" . " & " . "{$poke['type2']}</p>";
```
Comment (Roomsycakes01) [inline collectionchallenge.php]: single quotes on &

File: collectionchallenge.php [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    <!--	<link rel="stylesheet" href="normalize.css" />-->
+    <!--	<link rel="stylesheet" href="xxx.css" />-->
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+
+</head>
+<body>
+<h1>Pokemonz</h1>
+<?php
+foreach ($pokemon as $poke) {
+    echo "<div>";
```
Comment (Roomsycakes01) [inline collectionchallenge.php]: single quotes

File: collectionchallenge.php [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    echo "<h2>{$poke['name']}</h2>";
+    if ($poke['type2'] === "") {
+        echo "<p>{$poke['type1']}</p>";
+    } else {
+        echo "<p>{$poke['type1']}" . " & " . "{$poke['type2']}</p>";
+    }
+    echo "<ul>";
```
Comment (Roomsycakes01) [inline collectionchallenge.php]: single quotes

File: collectionchallenge.php [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    echo "<li>HP: {$poke['hp']}</li>";
+    echo "<li>Attack: {$poke['attack']}</li>";
+    echo "<li>Defense: {$poke['defense']}</li>";
+    echo "<li>SP. Attack: {$poke['spAttack']}</li>";
+    echo "<li>SP. Defense: {$poke['spDefense']}</li>";
+    echo "<li>Speed: {$poke['speed']}</li>";
+    echo "</div>";
```
Comment (Roomsycakes01) [inline collectionchallenge.php]: single quotes

File: collectiondb.php [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+$db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+$getPokemon = $db->prepare("SELECT `stats`.`name`, `stats`.`type1`, `stats`.`type2`, `stats`.`hp`, `stats`.`attack`, `stats`.`defense`, `stats`.`spAttack`, `stats`.`spDefense`, `stats`.`speed` FROM `stats`;");
+$getPokemon->execute();
+$pokemon = $getPokemon->fetchAll();
+
+
+if (isset($_POST['name'])) {
+    //if one type, add 'N/A' for type 2
+    $insertNewPokemon = $db->prepare("INSERT INTO `stats` (`name`,`type1`,`type2`, `hp`, `attack`, `defense`, `spAttack`, `spDefense`, `speed`) VALUES (:name, :type1, :type2, :hp, :attack, :defense, :spAttack, :spDefense, :speed);");
+    $insertNewPokemon->bindParam(':name', $_POST['name']);
```
Comment (Roomsycakes01) [inline collectiondb.php]: maybe just bind the parameters in the execute

File: collectiondb.php [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+    $insertNewPokemon->bindParam(':type2', $_POST['type2']);
+    $insertNewPokemon->bindParam(':hp', $_POST['hp']);
+    $insertNewPokemon->bindParam(':attack', $_POST['attack']);
+    $insertNewPokemon->bindParam(':defense', $_POST['defense']);
+    $insertNewPokemon->bindParam(':spAttack', $_POST['spAttack']);
+    $insertNewPokemon->bindParam(':spDefense', $_POST['spDefense']);
+    $insertNewPokemon->bindParam(':speed', $_POST['speed']);
+    $insertNewPokemon->execute();
+    header("Location: collectionchallenge.php");
+}
```
Comment (Roomsycakes01) [inline collectiondb.php]: newline

File: collectiondb.php [hunk 12]
```diff
@@ -0,0 +1,24 @@
+<?php
+
+$db = new PDO('mysql:host=db; dbname=pokemon', 'root', 'password' );
+$db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+$getPokemon = $db->prepare("SELECT `stats`.`name`, `stats`.`type1`, `stats`.`type2`, `stats`.`hp`, `stats`.`attack`, `stats`.`defense`, `stats`.`spAttack`, `stats`.`spDefense`, `stats`.`speed` FROM `stats`;");
+$getPokemon->execute();
+$pokemon = $getPokemon->fetchAll();
+
+
```
Comment (Roomsycakes01) [inline collectiondb.php]: too much whitespace

File: addpokemon.php (hunk 1 above)
Comment (EleanorRoberts) [reply addpokemon.php]: Darn I forgot, thanks!

File: collectionchallenge.php (hunk 8 above)
Comment (EleanorRoberts) [reply collectionchallenge.php]: Inside double quotes

File: collectionchallenge.php (hunk 9 above)
Comment (EleanorRoberts) [reply collectionchallenge.php]: Inside double quotes

File: collectiondb.php (hunk 10 above)
Comment (EleanorRoberts) [reply collectiondb.php]: Have updated :)

File: collectiondb.php (hunk 12 above)
Comment (EleanorRoberts) [reply collectiondb.php]: Removed

File: addpokemon.php [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+<head>
+    <meta charset="UTF-8" />
+    <title>Pokemon Collection</title>
+	<link rel="stylesheet" href="normalize.css" />
+	<meta name="viewport" content="width=device-width, initial-scale=1.0" />
+</head>
+<body>
+<h1>Add Pokemonz</h1>
+<form method="post">
+    <label for="name">Name <input type="text" name="name" required /></label>
```
Comment (jesshamby) [inline addpokemon.php:16]: Not sure what is best practice but I've only ever seen inputs outside of label tags: e.g. 
<label for="thing >Thing</label>
<input name="thing" type="" />

File: collectionFunctions.php [hunk 14]
```diff
@@ -0,0 +1,47 @@
+<?php
+
+/** Displays each pokemon and stats
+ * @param array $pokemon
+ * @return String
+ */
+function displayPoke(Array $pokemon) : String {
```
Comment (jesshamby) [inline collectionFunctions.php]: Picky but PSR-12 likes type-hinting to be a certain way (had this flagged by Ash myself): colon next to bracket 

https://gist.github.com/michaelcullum/c025f3870c9ea1dd2668#file-returntypesspacing-php

File: collectionFunctions.php [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+        $output .= "<li>Sp. Attack: {$poke['spAttack']}</li>";
+        $output .= "<li>Sp. Defense: {$poke['spDefense']}</li>";
+        $output .= "<li>Speed: {$poke['speed']}</li>";
+        $output .= "</div>";
+    }
+    return $output;
+}
+
+/** Connects to a database, determined by db name
+ * @param $database
```
Comment (jesshamby) [inline collectionFunctions.php]: add in the type hinting for this param

**Review by jesshamby — COMMENTED** [review]
Looks awesome - just a few small picky changes :)

File: collectionFunctions.php (hunk 15 above)
Comment (EleanorRoberts) [reply collectionFunctions.php]: Added type hinting to doc block

File: collectionFunctions.php (hunk 14 above)
Comment (EleanorRoberts) [reply collectionFunctions.php]: Thanks! Have updated now

File: addpokemon.php (hunk 13 above)
Comment (EleanorRoberts) [reply addpokemon.php:16]: Best practice is to have input inside label tag

**Pull request merged**